compiler/rustc_const_eval/src/transform/check_consts/check.rs

```diff
@@ -619,9 +619,12 @@ impl<'tcx> Visitor<'tcx> for Checker<'_, 'tcx> {
                 if base_ty.is_unsafe_ptr() {
                     if place_ref.projection.is_empty() {
                         let decl = &self.body.local_decls[place_ref.local];
-                        if let LocalInfo::StaticRef { def_id, .. } = *decl.local_info() {
-                            let span = decl.source_info.span;
-                            self.check_static(def_id, span);
```

Comment:
We already check_static each operand we see, so I don't understand why this is checked again here.